sdk/spring/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/repository/support/SimpleReactiveCosmosRepository.java

```diff
@@ -244,18 +254,38 @@ public Mono<Void> deleteAllById(Iterable<? extends K> ids) {
     public Mono<Void> deleteAll(Iterable<? extends T> entities) {
         Assert.notNull(entities, "The given Iterable of entities must not be null!");
 
-        return Flux.fromIterable(entities).flatMap(this::delete).then();
+        if (entityInformation.getPartitionKeyFieldName() != null) {
+            Flux<CosmosItemOperation> cosmosItemOperationFlux = Flux.fromIterable(entities).map(entity -> {
+                K id = entityInformation.getId(entity);
+                return CosmosBulkOperations.getDeleteItemOperation(id != null ? id.toString() : "",
```

Comment:
I don't believe it can ever be null, but the linting report gave me a lot of grief with using entityInformation.getId() and this the only way I could get it to be happy.